invenio_search/walkers/elasticsearch_no_keywords.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015, 2016 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Implement AST convertor to Elastic Search DSL."""
+
+from invenio_query_parser.ast import (
+    AndOp, DoubleQuotedValue, EmptyQuery,
+    GreaterEqualOp, GreaterOp, Keyword,
+    KeywordOp, LowerEqualOp, LowerOp,
+    NotOp, OrOp, RangeOp, RegexValue,
+    SingleQuotedValue, Value, ValueQuery
+)
+from invenio_query_parser.visitor import make_visitor
+
+
+class KeywordAppearance(Exception):
+    pass
+
+
+class ElasticSearchNoKeywordsDSL(object):
+
+    """Implement visitor to create Elastic Search DSL for queries that don't include keywords."""
+
+    visitor = make_visitor()
+
+    def __init__(self):
+        self.values = ""
+
+    @visitor(KeywordOp)
+    def visit(self, node, left, right):
+        raise KeywordAppearance()
+
+    @visitor(AndOp)
+    def visit(self, node, left, right):
+        return {
+            "multi_match": {
+                "message": {
+                    "query": self.values,
+                    "operator": "or",
+                    "zero_terms_query": "all",
+                    "fields": [
+                        "title^3",
+                        "title.raw^10",
+                        "abstract^2",
+                        "abstrac.raw^4",
+                        "author^10",
+                        "author.raw^15",
+                        "reportnumber^10",
+                        "eprint^10",
+                        "doi^10"]}}}
```

Comment:
Yes most probably it will crash. I will check.
I copied the query from https://github.com/inspirehep/inspire-next/issues/682, I'll do some tweaking to the values to see the results and come back with more info .